src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts

```diff
@@ -246,6 +266,172 @@ export class InstructorCoursesPageComponent implements OnInit {
     });
   }
 
+  /**
+   * Creates a copy of a course including the selected sessions.
+   */
+  onCopy(courseId: string, courseName: string, timeZone: string): void {
+    if (!courseId) {
+      this.statusMessageService.showErrorToast('Course is not found!');
+      return;
+    }
+
+    this.feedbackSessionsService.getFeedbackSessionsForInstructor(courseId).subscribe((response: FeedbackSessions) => {
+      const modalRef: NgbModalRef = this.ngbModal.open(CopyCourseModalComponent);
+      modalRef.componentInstance.oldCourseId = courseId;
+      modalRef.componentInstance.oldCourseName = courseName;
+      modalRef.componentInstance.newTimeZone = timeZone;
+      modalRef.componentInstance.courseToFeedbackSession[courseId] = response.feedbackSessions;
+      modalRef.componentInstance.selectedFeedbackSessions = new Set(response.feedbackSessions);
+      modalRef.result.then((result: CopyCourseModalResult) => this.createCourse(result), () => {});
+    }, (resp: ErrorMessageOutput) => {
+      this.statusMessageService.showErrorToast(resp.error.message);
+    });
+  }
+
+  /**
+   * Creates a new course with the selected feedback sessions
+   */
+  createCourse(result: CopyCourseModalResult): void {
+    this.isCopyingCourse = true;
+    this.numberOfSessionsCopied = 0;
+    this.totalNumberOfSessionsToCopy = result.totalNumberOfSessions;
+    this.copyProgressPercentage = 0;
+
+    this.courseService.createCourse({
+      courseName: result.newCourseName,
+      timeZone: result.newTimeZone,
+      courseId: result.newCourseId,
+    })
+    .pipe(
+      finalize(() => {
+        this.courseService
+          .getCourseAsInstructor(result.newCourseId)
+          .subscribe((course: Course) => {
+            this.activeCourses.push(this.getCourseModelFromCourse(course));
+            this.activeCoursesDefaultSort();
+            this.isCopyingCourse = false;
```

Comment:
Btw, when you are copying feedback sessions, is it also considered `CopyingCourse`?
My doubt would be is it possible for the same create course to be fired the second time while the first is still creating sessions, which would result in 2x or 3x (depends on how many times the course is copied) and an unjustifiable backend load?

Reply:
Copying feedback sessions is a part of `CopyingCourse`. When feedback sessions are still being copied, the copy button will be disabled so instructors cannot copy another course at the same time. But they can still add a new course with just course name and course id.